services/src/main/resources/org/keycloak/protocol/oidc/endpoints/login-status-iframe.html

```diff
@@ -55,7 +55,12 @@
           init = { clientId, origin };
         }
 
-        const cookie = getSessionCookie();
+        var cookie = getSessionCookie();
+
+        // Cookie value may be wrapped in quotes
+        if (cookie != null && cookie.startsWith('"') && cookie.endsWith('"')) {
+          cookie = cookie.substr(1, cookie.length - 2);
+        }
```

Comment:
I think it would make more sense to handle this a bit lower level in the `getCookieByName()` function, where the value is always defined, also:

- It is preferred to use [strict inequality](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_inequality) (e.g. `!==`) most of the time for correctness reasons.
- The `substr()` method [is deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr), but you can also use [`slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) to accomplish this.

Putting that together you should get something like this:

```js
function getCookieByName(name) {
  const cookies = new Map();

  for (const cookie of document.cookie.split(";")) {
    const [key, value] = cookie.split("=").map((value) => value.trim());
    // The value might be wrapped in quotes, so remove them.
    cookies.set(key, value.startsWith('"') && value.endsWith('"') ? value.slice(1, -1) : value);
  }

  return cookies.get(name) ?? null;
}
```

Reply:
Updated